### PR TITLE
[dav1d] Enable fuzzing on i386 architecture

### PR DESCRIPTION
**Pull request by tysmith**

### CODE REVIEW

**Comment by tysmith** [comment]
@jannau do you have any feedback before merging?

File: projects/dav1d/build.sh [hunk 1]
```diff
@@ -23,8 +23,13 @@ rm -rf ${build}
 mkdir -p ${build}
 
 # build library
+if [ "$ARCHITECTURE" = "x86_64" ]; then
+	use_asm=true
+else
+	use_asm=false
+fi
```
Comment (jannau) [inline projects/dav1d/build.sh]: Is this necessary? The main difference between x86_64 and i386 would be in the asm. If it is necessary to build i386 I would prefer to fix it instead. Is there information available on how oss-fuzz builds for i386? Does it just adds `-m32` to CFLAGS/LFLAGS? I'll check tonight (central european summer time).

File: projects/dav1d/Dockerfile [hunk 2]
```diff
@@ -20,7 +20,9 @@ MAINTAINER janne-vlc@jannau.net
 ADD bionic.list /etc/apt/sources.list.d/bionic.list
 ADD nasm_apt.pin /etc/apt/preferences
 
-RUN apt-get update && apt-get install -y curl python3-pip nasm && \
+RUN dpkg --add-architecture i386 && \
+    apt-get update && \
+    apt-get install -y curl libstdc++-5-dev libstdc++-5-dev:i386 python3-pip nasm && \
```
Comment (jannau) [inline projects/dav1d/Dockerfile]: Is `libstdc++-5-dev ` required for building the fuzzing engine or for for linking the final fuzzing binary?

Comment (jannau) [reply projects/dav1d/Dockerfile]: I see it is for some reason required after adding i386 support

File: projects/dav1d/build.sh (hunk 1 above)
Comment (jannau) [reply projects/dav1d/build.sh]: So it only adds `-m32` to CFLAGS. this doesn't work for meson. I'll open a merge request with meson "cross" compiling support  to supecede this MR.

**Comment by jannau** [comment]
See #2824

**Comment by tysmith** [comment]
That looks great, thanks!

**Pull request closed without merging**